overlord/snapstate/handlers.go

```diff
@@ -3626,6 +3826,23 @@ func (m *SnapManager) undoRefreshAliases(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	setupAliasesFailedInPruneMode, err := checkSetupAliasesFailedInPruneMode(t, snapName)
+	if err != nil {
+		return err
+	}
+
+	// AliasesPending needs to be fixed if `setup-aliases` failed in prune mode.
+	//
+	// the following sequence triggers the edge-case:
+	//	1. `remove-aliases` skips removing aliases for refresh-app-awareness
+	//	    keeping AliasesPending as false
+	//	2. `set-auto-aliases` sets AliasesPending to true
+	//	3. `setup-aliases` fails mid-way before setting AliasesPending to false
+	if setupAliasesFailedInPruneMode {
```

Comment:
wouldn't it be easier to check a combination of snapst.AliasesPending && the predicate?